polytracker/mapping.py

```diff
@@ -73,49 +73,71 @@ def marker_to_ranges(self, m: bytearray) -> List[CavityType]:
                     ranges.append((start, i))
                     start = None
         if start is not None:
-            ranges.append((start, len(m) - 1))
+            ranges.append((start, len(m)))
         return ranges
 
     def file_cavities(self) -> Dict[Path, List[CavityType]]:
```

Comment:
Was this tested for equivalent performance? We had issues with improperly implemented DFS traversals before.

Reply:
@surovic  I believe we can resolve this after our discussion, right?
The same DFS is used as previously. Also a bug was uncovered and is solved by this new implementation. We would incorrectly mark source offsets as used in case of multiple source files. Have probably not been a problem in practice though.